components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -615,10 +672,36 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
         if let Some(v) = self.get32_by_fast_index(code_point) {
             v
         } else {
-            self.get32_by_small_index(code_point)
+            self.get32_by_small_index_cold(code_point)
+        }
+    }
+
+    /// Returns the value that is associated with `bmp` in this [`CodePointTrie`].
+    #[inline(always)]
+    pub fn get16(&self, bmp: u16) -> T {
+        // LLVM's optimizations have been observed not to be 100%
+        // reliable around collapsing away unnecessary parts of
+        // `get32`, so not just calling `get32` here.
+        let code_point = u32::from(bmp);
+        if let Some(v) = self.get32_by_fast_index(code_point) {
+            v
+        } else {
+            self.get32_by_small_index_cold(code_point)
         }
     }
 
+    /// Lookup trie value by non-Basic Multilingual Plane Scalar Value.
+    ///
+    /// The return value may be bogus (not necessarily `error_value`) is the argument is actually in
+    /// the Basic Multilingual Plane or above the Unicode Scalar Value
+    /// range (panics instead with debug assertions enabled).
+    #[inline(always)]
+    pub fn get_supplementary(&self, supplementary: u32) -> T {
```

Comment:
Should this be `get32_supplementary`? Or should the argument be `char`, which would require the caller to turn a `u32` to a `char`?